modules/saml/src/Controller/ServiceProvider.php

```diff
@@ -126,6 +135,20 @@ public function login(Request $request, string $sourceId): RunnableResponse
             'ReturnTo' => $httpUtils->checkURLAllowed($returnTo),
         ];
 
+        // Set options provided by the Initiator Request
+        // configuration.
+        if($spSource->isRequestInitiation()) {
+            if($request->query->has('entityID')) {
+                $options['saml:idp'] = $request->query->get('entityID');
+            }
+            if($request->query->has('isPassive')) {
+                $options['isPassive'] = $request->query->get('isPassive');
```

Comment:
Does this (and the forceauthn) need to be `->getBoolean()` so we have the correct type in `$options`?

Reply:
We could. The reason why i did it this way is because when the actual value is used it is type casted:
```php
        if (isset($state['isPassive'])) {
            $ar->setIsPassive((bool) $state['isPassive']);
        }
```
Better be strict though.